<launch>

    <!-- Pose selector mesh visualizer-->
    <arg name = "mesh_config" default="$(find pose_selector)/config/example_mesh_config.yaml" doc="Path to mesh objects config file"/>
    <!-- Remap pose_selector_get_all_poses service-->
    <remap from="/pose_selector_get_all_service" to="/pose_selector_node/pose_selector_get_all" />
    
    <node pkg="pose_selector" type="pose_selector_visualizer.py" name="pose_selector_visualizer_node" output="screen">
        <rosparam command="load" file="$(arg mesh_config)" />
    </node>

    <!-- Pose selector -->
    <arg name="objects_of_interest" value="[]"/>
    <arg name="config_file" value="pose_selector_visualizer_demo.yaml"/>
    <arg name="global_reference_frame" value="map" />

    <node pkg="pose_selector" type="pose_selector_node" name="pose_selector_node" output="screen">
        <rosparam command="load" file="$(find pose_selector)/config/$(arg config_file)" />
        <rosparam param="objects_of_interest" subst_value="True">$(arg objects_of_interest)</rosparam>
        <param name="global_reference_frame" type="str" value="$(arg global_reference_frame)"/>
    </node>

</launch>
